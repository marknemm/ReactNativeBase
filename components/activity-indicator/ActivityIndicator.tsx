import { ActivityIndicator as NativeActivityIndicator } from 'react-native';
import type { ActivityIndicatorProps } from './ActivityIndicator.interfaces';

/**
 * A component that displays an activity indicator or load spinner.
 *
 * @param props The {@link ActivityIndicatorProps}.
 * @returns The {@link ActivityIndicator} component.
 */
const ActivityIndicator: React.FC<ActivityIndicatorProps> = ({
  isVisible = true,
  ...activityIndicatorProps
}) =>
  isVisible && <NativeActivityIndicator {...activityIndicatorProps} />;

export default ActivityIndicator;
export type * from './ActivityIndicator.interfaces';
