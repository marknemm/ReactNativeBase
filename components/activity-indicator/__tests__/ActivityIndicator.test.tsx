import ActivityIndicator from '@components/activity-indicator/ActivityIndicator';
import AppProvider from '@test/contexts/app/AppProvider';
import { render, screen } from '@testing-library/react-native';

describe('<ActivityIndicator />', () => {
  describe('snapshots', () => {
    it('renders correctly', () => {
      render(
        <ActivityIndicator />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly when not visible', () => {
      render(
        <ActivityIndicator isVisible={false} />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
