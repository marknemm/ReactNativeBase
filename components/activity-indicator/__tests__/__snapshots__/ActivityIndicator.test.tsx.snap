// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivityIndicator /> snapshots renders correctly 1`] = `<ActivityIndicator />`;

exports[`<ActivityIndicator /> snapshots renders correctly when not visible 1`] = `null`;
